src/plugins/mf_priority.cpp

```diff
@@ -204,6 +212,56 @@ static int validate_cb (flux_plugin_t *p,
                                     "user does not belong to specified bank");
     }
 
+    mj_it = users_mj.find (userid);
+
+    mj_inner_it = (bank == NULL) ?
+        mj_it->second.find ("default") :
+        mj_it->second.find (bank);
+
+    if (mj_inner_it->second[1] >= mj_inner_it->second[0])
+        return flux_jobtap_reject_job (p, args,
+                                    "user has max number of jobs submitted");
+
+    // increment user's max_jobs count
+    mj_inner_it->second[1]++;
```

Comment:
It looks like there may be  a problem, though I could be misunderstanding the code since I'm not familiar with C++ `std::map`:

It _appears_ that current job counts are tracked separately for a default bank "name" versus literally the "default" bank. E.g. if a user has a default bank of "A" with a limit of 4 jobs, and submits 4 jobs to bank "A", then submits a job with no bank, then the "default" bank max-jobs will be checked which is currently zero, so the job will be accepted.

Sorry if I misunderstood the code.

In any event, I'd suggest a test to ensure that jobs submitted with a default bank definitely have the default bank limit applied when jobs have a mix of explicit bank versus no bank.

Reply:
Ah, yes. I think you are right. Two separate counts would be tracked in both the "default" key-value entry and the bank "name" key-value entry. So I'll need to re-evaluate my approach, and probably just keep one count of active jobs, regardless if the user is submitting under a default bank or explicitly setting their default bank with the bank "name."

Reply:
What I would suggest is to have a map of userid->default_bank. When a bank is not provided, then you look up the default bank name and use that name in place of a provided bank name. Everything else then works the same, except there is never a `userid,"default"` user/bank pair.

One gotcha will be if the user's default bank changes between when a job starts and when it ends. You can't look up the default bank name at the end of the job from `default_banks[userid]` since it will have changed, and you will decrement the wrong entry. So, the default bank for a job should be somehow set  in the `job.validate` callback, and used for that job from that point on.

You could do this at first by stashing the bank name in the job "aux" hash, i.e. `flux_jobtap_job_aux_set()`. If you do this for all cases (i.e. whether a bank is set of your are looking up the default bank), then in later callbacks you can fetch the bank name directly with `flux_jobtap_job_aux_get()`. Or, you could instead stash a "bank_struct" with information like:
```
struct bank_info {
    const char *name;
    double fairshare;
    int max_jobs;
    int current_jobs;
};
```

Then your `inactive_cb` could be simplified to something like:
```c
static int inactive_cb (flux_plugin_t *p,
                        const char *topic,
                        flux_plugin_arg_t *args,
                        void *data)
{   
    struct bank_info *bank;
    
    if ((bank = flux_jobtap_job_aux_get (p,
                                         FLUX_JOBTAP_CURRENT_JOB,
                                         "mf_priority:bank")))
        bank->current_jobs--;
    
}
```

Reply:
> What I would suggest is to have a map of userid->default_bank. When a bank is not provided, then you look up the default bank name and use that name in place of a provided bank name. Everything else then works the same, except there is never a `userid,"default"` user/bank pair.

Thanks for this suggestion. I think I understand what you are suggesting. With this change, I'm able to test the use case of submitting a mix of both "explicitly-set bank" and "default bank" jobs that both contribute to one active job count with the following:

```shell
test_expect_success 'submit max number of jobs with a mix of default bank and explicity set bank' '
	jobid1=$(flux python ${SUBMIT_AS} 5011 sleep 60) &&
	jobid2=$(flux python ${SUBMIT_AS} 5011 --setattr=system.bank=account3 sleep 60) &&
	jobid3=$(flux python ${SUBMIT_AS} 5011 sleep 60)
'

test_expect_success 'submit a job while already having max number of active jobs' '
	test_must_fail flux python ${SUBMIT_AS} 5011 sleep 60 > max_jobs.out 2>&1 &&
	test_debug "cat max_jobs.out" &&
	grep "user has max number of jobs submitted" max_jobs.out &&
'
```

I can push up a fix that contains this change. 

> You could do this at first by stashing the bank name in the job "aux" hash, i.e. flux_jobtap_job_aux_set(). If you do this for all cases (i.e. whether a bank is set of your are looking up the default bank), then in later callbacks you can fetch the bank name directly with flux_jobtap_job_aux_get(). 

I'm sorry that I don't understand this enough, so this is probably a dumb question. The concept makes sense to me; in the case where a user's default bank changes during the course of an active job, we'll need to correctly decrement the appropriate bank. Is it possible to only pass the bank name with `flux_jobtap_job_aux_set ()` and `get ()`? Is there a reason for passing other information like `fairshare` and `max jobs` through a struct?

Reply:
> Is there a reason for passing other information like fairshare and max jobs through a struct?

The only reason is that this avoid the extra lookup into the user/bank map to find the current job count. If you instead kept a `bank_info` structure in the map instead of raw counts, then you can attach a _reference_ to the bank information for that job to the job with `job_aux_set()`. That may help you down the line, (e.g. it will avoid the extra lookup when reprioritizing jobs as well).

Storing the bank name will work just as well and this was only a suggestion for another way to do it.

Reply:
Ah, okay. I could see this also reducing the extra lookup for fetching the bank values for calculating job priority as well.

Would you be okay with solely passing in the default bank information through `job_aux_set ()` and `get ()` in this PR, and posting a follow-up refactoring PR to use a `bank_info` structure to improve the information passing for both the max jobs limit and the priority calculation?

Reply:
Sure! I was just suggesting another way to do it.